Automattic/newspack-theme
# Removes remaining original menu cruft 🔥

### All Submissions:


### Changes proposed in this Pull Request:

This is hopefully the final PR cleaning up unneeded menu code:

* It re-removes the `newspack_add_ellipses_to_nav()` function, something I'd previously removed but accidentally reintroduced in another PR (I had also deleted it and re-added it in that PR's history, and poorly fixed that merge conflict by re-adding it). 
* It removes `newspack_add_mobile_parent_nav_menu_items()`, a function that duplicated each 'parent' menu item, for the mobile navigation.
* It removes some code that added some markup and an SVG link to the above menu items.
* It removes the entire touch-keyboard-navigation.js. Much of this code is related to the original priority menu, and though we will need some JavaScript for the non-AMP version of the mobile menu (to toggle it open and closed on small screens, and to make sure the dropdown menus can still be used on larger tablets), it will be a near-full rewrite of what's there. It seemed easier to remove, and start fresh. 

### How to test the changes in this Pull Request:

1. Set up a primary menu with at least one dropdown.
2. On a touch device, try tapping on the `V` icon next to the menu link with a dropdown -- it should open a full-screen version of that dropdown (with some terrible styles):

![Screen Shot 2019-08-05 at 3 13 02 PM](https://user-images.githubusercontent.com/177561/62498291-8fabfe80-b793-11e9-851c-cf77f3628252.png)

![image](https://user-images.githubusercontent.com/177561/62498266-7e62f200-b793-11e9-8992-a24167a0546e.png)

3. Apply the PR.
4. Make sure the menus display as expected on larger screens. 
5. On a touch device, try tapping on the `V` icon next to a menu link with a dropdown again -- it should now open the menu as a full-width dropdown:

![image](https://user-images.githubusercontent.com/177561/62498434-ffba8480-b793-11e9-872b-0fa244989644.png)

(Mobile header styles are in progress so please don't worry too much about how this all looks right now!). 

### Other information:
